script.js

```diff
@@ -1,4 +1,489 @@
+    }
+    /// draw
+    if (p1StartHand == pciHand) {
+      drawCounter = drawCounter + 1;
+      return `<b>${playerName}'s final hand = ${p1StartHand}<br>Computer's final hand = ${pciHand}.<br><br>Since both player have the same hand, therefore its a draw!<br><br>${playerName} has won ${p1winCounter} round<br>Computer has won ${pcWinCounter} round<br>Draw for ${drawCounter} round<br><br>Click the submit button to play ♠ again!`;
+    }
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  myOutputValue = ``;
+  if (gameMode == "welcome") {
+    gameMode = "playerName";
+    return `<b>Welcome to blackjack! Please enter a username!`;
+  }
+  if (gameMode == "playerName") {
+    if (input == playerName) {
+      return `<b>Please enter an username!`;
+    }
+    playerName = input;
+    gameMode = "Bet";
+    return `<b>Hey ${playerName}, decide how many chips you would like to bet in this round,<br>by typing the number of chips in the long white space<br><br>DO NOT SPELL ELSE THE BETTING SYSTEM WILL CRASH!<br><br>You currently have ${playingChips} chips left<br><br>Example..if you have bet 20 chips and you win, you'll have 120 chips<br><br>Otherwise you'll have 80 chips left<br>In an event of a draw, nothing will happen<br><br>You have 100 chips initially<br><br>Click on the submit button to confirm your bet and receive your starting hand!`;
+  }
+  if (gameMode == "betAgain") {
+    gameMode = "Bet";
+    playerBets = input;
+    return `<b>Hey ${playerName}, decide how many chips you would like to bet in this round,<br>by typing the number of chips in the long white space<br><br>DO NOT SPELL ELSE GAME WILL CRASH!<br><br>You currently have ${playingChips} chips left<br><br>Example..if you have bet 20 chips and you win, you'll have 120 chips<br>Otherwise you'll have 80 chips left<br><br><br><br>Click on the submit button to confirm your bet and receive your starting hand!`;
+  }
+  if (gameMode == "Bet") {
+    playerBets = input;
+    gameMode = "p1";
+    return `<b>You have bet ${playerBets} chips !`;
+  }
+  if (gameMode == "chooseAce") {
+    if (!(input == 1 || input == 11)) {
+      return `<b>♤ You have entered an invalid entry ❌!<br>Your current hand is ${p1StartHand}<br><br>Please only enter 1 or 11!`;
+    }
+    myOutputValue = aceOnHand(input);
+    gameMode = "aceHOS";
+    return myOutputValue;
+  }
+  if (gameMode == "aceHOS") {
+    if (!(input == "hit" || input == "stand")) {
+      return HOSinvalid;
+    }
+    myOutputValue = hitOrStand(input);
+    return myOutputValue;
+  }
+  if (gameMode == "p1") {
+    myOutputValue = randomCard();
+    return myOutputValue;
+  }
+  if (gameMode == "p1HOS") {
+    if (!(input == "hit" || input == "stand")) {
+      return HOSinvalid;
+    }
+    myOutputValue = hitOrStand(input);
+    return myOutputValue;
+  }
```

Comment:
whats the differnece betwen this mode and aceHOS, the code inside look duplicated. can this be made into a function?

Reply:
thanks for helping me realise this i fixed it already